### PR TITLE
 Fix crash in JSON schema validation when additional properties are provided (fixes #548)

### LINKED ISSUE #548: 500 error when json validation fails with additional properties
`<ValidationError: "Additional properties are not allowed ('blockID', 'id' were unexpected)">`

```
  File "/home/mathieu/Code/Mozilla/cliquet/cliquet/resource/__init__.py", line 395, in put
    new_record = self.process_record(post_record, old=existing)
  File "/home/mathieu/Code/Mozilla/kinto/kinto/views/records.py", line 71, in process_record
    field = e.path.pop() if e.path else e.validator_value.pop()
AttributeError: 'bool' object has no attribute 'pop' lang=None uid=67f1f8fc8d3dd228096950c20d6515c4786a675264741bec2995a560519aad87

```

Possible fix:

```diff
         except jsonschema_exceptions.ValidationError as e:
-            field = e.path.pop() if e.path else e.validator_value.pop()
+            try:
+                field = e.path.pop() if e.path else e.validator_value.pop()
+            except AttributeError:
+                field = None
```

### PR DESCRIPTION
**Pull request by leplatrem**
Fixes #548 
Based on #554

### CODE REVIEW

**Comment by leplatrem** [comment]
@Natim r?

**Comment by Natim** [comment]
Please rebase to ease the review :+1:

**Comment by leplatrem** [comment]
Done

File: kinto/tests/test_views_collections_schema.py [hunk 1]
```diff
@@ -212,3 +212,12 @@ def test_records_are_validated_on_patch(self):
...
                             headers=self.headers)
+
+    def test_additional_properties_are_rejected(self):
+        record_id = '5443d83f-852a-481a-8e9d-5aa804b05b08'
+        record = VALID_RECORD.copy()
+        record['extra'] = 'blah!'
+        self.app.put_json('%s/%s' % (RECORDS_URL, record_id),
+                          {'data': record},
+                          headers=self.headers,
+                          status=400)
```
Comment (Natim) [inline kinto/tests/test_views_collections_schema.py]: I would also validate that the error message give enough feedback for the user to solve the issue.

**Comment by Natim** [comment]
r+

**Pull request merged**